Mysql_db single_transaction and quick

hyperized opened the pull request:
##### Issue Type:
- Feature Pull Request
##### Plugin Name:

mysql_db
##### Summary:

Adding functionality for 'single_transaction' and 'quick' to aid fast dumping in large (InnoDB) databases.
##### Example:

```
- mysql_db:
    state: dump
    name: database
    target: dump.sql.gz
    single_transaction: true
    quick: true
```

Commit:
Add support for provider network options.

Adds new parameters for specifying options for provider network plugins.

resmo commented:
Thanks @hyperized. To the current maintainers, @Jmainguy please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

resmo commented:
@hyperized there are merge commits in this PR. Could you please rebase it?

Jmainguy commented:
Code looks pretty good, I just need to test it. Will feedback once I have done that

hyperized commented:
@resmo I'm not very familiar with rebasing, as you can see I usually merge things. Can you point me in the right direction? I know the commands, but where do I rebase onto? d9968b0, its parent, other?

Commit:
Mysql_db single_transaction and quick

hyperized commented:
@resmo rebased :)

gregdek commented:
Thanks @hyperized for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

hyperized commented:
@gregdek please review, its been rebased :)

gregdek commented:
Thanks @hyperized. To the current maintainers, @Jmainguy please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

rcarrillocruz committed:
Only check default_project on resource creation

The default_project is checked at the beginning of the module.
This raises an exception if the project passed does not exist.
This logic only makes sense on resource creation, if a user
puts state=absent the module fails, even though the default
project is not relevant

Jmainguy commented on database/mysql/mysql_db.py:
```diff
@@ -268,8 +286,7 @@ def main():
     try:
-        cursor = mysql_connect(module, login_user, login_password, config_file, ssl_cert, ssl_key, ssl_ca,
-                               connect_timeout=connect_timeout)
+        cursor = mysql_connect(module, login_user, login_password, config_file, ssl_cert, ssl_key, ssl_ca, connect_timeout=connect_timeout)
```
looked better before, suggest changing it back

Jmainguy commented on database/mysql/mysql_db.py:
```diff
@@ -328,3 +345,5 @@ def main():
 if __name__ == '__main__':
     main()
+
+
```
no need for extra blank lines, you can delete these 2

Jmainguy commented:
tested and works. Make the small changes above and we can shipit

gregdek commented:
Thanks again to @hyperized for this PR, and thanks @Jmainguy for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

gregdek commented:
@Jmainguy don't say "shipit" when you mean "if x, then shipit". The bot isn't smart enough to understand the difference. :)

hyeonseongc committed:
added rsync protocol support (#1999)

* added rsync protocol support

* correction for example document(add example for push on delegate)

* use startswith method for safety

sivel committed:
overwrite option was added for 2.1

resmo committed:
openstack: doc fix, add version_added

see #3388

resmo commented:
I see some potential problems, because these new args are booleans, they should be used as booleans:

E.g.

``` yaml
- mysql_db:
    state: dump
    name: database
    target: dump.sql.gz
    single_transaction: true
    quick: false
```

As quick is not None it would be used, this wouldn't be expected. 

See comments inline.

resmo commented on database/mysql/mysql_db.py:
```diff
@@ -226,6 +240,8 @@ def main():
             config_file=dict(default="~/.my.cnf"),
+            single_transaction=dict(default=None),
+            quick=dict(default=None),
         ),
```
use bool here

``` python
single_transaction=dict(default=False, type='bool'),
quick=dict(default=False, type='bool'),
```

resmo commented on database/mysql/mysql_db.py:
```diff
@@ -124,6 +134,10 @@ def db_dump(module, host, user, password, db_name, target, all_databases, port,
         cmd += " --all-databases"
     else:
         cmd += " %s" % pipes.quote(db_name)
+    if single_transaction is not None:
```
test for True

``` python
if single_transaction:
```

resmo commented on database/mysql/mysql_db.py:
```diff
@@ -124,6 +134,10 @@ def db_dump(module, host, user, password, db_name, target, all_databases, port,
         cmd += " %s" % pipes.quote(db_name)
+    if single_transaction is not None:
+        cmd += " --single-transaction=true"
+    if quick is not None:
```
test for True

``` python
if quick:
```

resmo commented on database/mysql/mysql_db.py:
```diff
@@ -56,6 +56,16 @@
+    description:
+      - Execute the dump in a single transaction
+    required: false
+    default: null
```
set `default: false` and also add `version_added: "2.1"`

resmo commented on database/mysql/mysql_db.py:
```diff
@@ -56,6 +56,16 @@
+    description:
+      - Option used for dumping large tables
+    required: false
+    default: null
```
set `default: false` and also add `version_added: "2.1"`

hyperized commented:
Awesome, so no shipit after all. I'd hope for a little more consistency in the future review process. I'll try to get all fixes in today.

resmo commented:
@hyperized Thanks for your feedback. I don't like it to see you unhappy how it went on this PR, my apologize.

The core and core community are working hard to continuing improving the open PR handling, but at the end it is the code and the users counts the most, to achieve a consistent ansible experience.

Thanks for your understanding, set 'ready_for_review' if the works are done, I'll handle it fast tracked.

hyperized commented:
Meh, I dont know git as well as I should. Have a little patience with me, sorry.

hyperized commented:
Derp, I'll make a new PR :)

Jmainguy commented:
@hyperized I should have caught all the things @resmo caught during my review. So I apologize for not catching them during my review, and I will do better in future reviews. Thanks again for submitting this code.
